assets/js/copy-code-blocks.js

```diff
@@ -0,0 +1,40 @@
+// copy-code-blocks.js
+// Copyright: 2023  Scientific Python Developers
+// Author: Adam Porter <adam@alphapapa.net>
+// License: BSD 3-Clause License
+//   see <https://raw.githubusercontent.com/scientific-python/scientific-python-hugo-theme/main/LICENSE>
+
+// This file implements a button attached to code blocks which copies
+// the code to the clipboard.
+
+// This selector should match the PRE elements, each of which may
+// itself contain the text to be copied, or may contain a CODE element
+// which contains the text to be copied.
+const pre_selector = "div.highlight pre";
+
+document.querySelectorAll(pre_selector).forEach(add_button_to);
+
+function add_button_to(element) {
+    button = document.createElement("button");
+    button.classList.add("copy-button")
+    button.append(document.createTextNode("Copy"));
+    button.addEventListener("click", copy_content_of);
```

Comment:
This should be modeled after PDST, i.e. use a FontAwesome icon, and then when clicked display the Copied! text.

https://pydata-sphinx-theme.readthedocs.io/en/stable/examples/kitchen-sink/blocks.html

Fading effects are not required.

Reply:
That does look nice.

I implemented something a bit simpler to start with, please see https://github.com/scientific-python/scientific-python-hugo-theme/pull/418/commits/26ad2f903664d285b9db33702d82890e1708a9dd and https://github.com/scientific-python/scientific-python-hugo-theme/pull/418/commits/401e4c7141e0e50e3ea64f56b8fa960f110723fe and let me know if you'd like more changes to it.